src/storage.ts

```diff
@@ -0,0 +1,94 @@
+const store = (type: 'sessionStorage' | 'localStorage') => ({
+	available: undefined as boolean | undefined,
+	storage: window[type],
+	isAvailable(): boolean | undefined {
```

Comment:
Under what circumstances does this return `undefined`? Isn't it always `boolean`?

Reply:
I agree. To change this though would change the contract we're exposing which could pose problems for frontend. My goal was to try to keep the same api. But yeah, `isAvailable(): boolean {` is a lot nicer

Reply:
Okay, looking at this more I see that `undefined` does make sense. When `isAvailable === undefined` it means the check has not been made yet, it is litterally undefined. So this is correct.